jsonassert-shade/pom.xml

```diff
@@ -0,0 +1,136 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd"
+		 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+	<modelVersion>4.0.0</modelVersion>
+	<artifactId>jsonassert-shade</artifactId>
+	<version>0.4.15-SNAPSHOT</version>
+
+	<parent>
+		<groupId>com.toomuchcoding.jsonassert</groupId>
+		<artifactId>jsonassert-parent</artifactId>
+		<version>0.4.15-SNAPSHOT</version>
+		<relativePath>..</relativePath>
+	</parent>
+
+	<properties>
+		<maven-shade-plugin.version>3.2.1</maven-shade-plugin.version>
+		<maven-dependency-plugin.version>3.1.1</maven-dependency-plugin.version>
+		<build-helper-maven-plugin.version>3.0.0</build-helper-maven-plugin.version>
+	</properties>
+
+	<dependencies>
+		<dependency>
+			<groupId>${project.groupId}</groupId>
+			<artifactId>jsonassert</artifactId>
+			<version>${project.version}</version>
+			<scope>compile</scope>
+		</dependency>
+	</dependencies>
+
+	<build>
+		<plugins>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-shade-plugin</artifactId>
+				<version>${maven-shade-plugin.version}</version>
+				<executions>
+					<execution>
+						<phase>package</phase>
+						<goals>
+							<goal>shade</goal>
+						</goals>
+						<configuration>
+							<createDependencyReducedPom>true</createDependencyReducedPom>
+							<shadedArtifactAttached>false</shadedArtifactAttached>
+							<createSourcesJar>true</createSourcesJar>
+							<shadeSourcesContent>true</shadeSourcesContent>
+							<keepDependenciesWithProvidedScope>false
+							</keepDependenciesWithProvidedScope>
+							<relocations>
+								<relocation>
+									<pattern>com.jayway</pattern>
```

Comment:
If we do not relocate `com.jayway` it will simplify things quite a bit. Really the issue is not `jayway` but its implementation details (`smart-json -> smart-asserter -> asm5` ).

Reply:
If we don't relocate it then spring cloud contract doesn't work ;)

Reply:
Hows that? I would expect `com.jayway` to stay the same and only its internal libs in question be relocated.

Reply:
there were broken tests in spring cloud contract cause internally `com.jayway` used the `shaded.` version of the lib but a non shaded was required. Something like that.

Reply:
@marcingrzejszczak 

I think we can make it work w/ a bit of adjustment in SCC. 

Take a peek at this [branch|https://github.com/bono007/spring-cloud-contract/commit/19dcc4843ac328cf8b79cd8549df55848614c711] commit. It uses a version of `jsonassert` w/o `com.jayway.JsonPath` shaded. 

A couple of points
- one of the specs uses `shaded.net.minidev.json.JSONArray` which is fine. 
- the KafkaStubMessages publicly exposes `json-smart (net.minidev.*)` via its imports even though its just used internally used to parse messages. We should replace this usage as exposing a shaded lib publicly negates the benefit of shading. 

The reason I am going on and on about not shading `com.jayway.JsonPath` is because its publicly exposed via SCC and almost guaranteed that consumers are referencing it in their apps - this will break them. 

All of the above aside though...

In https://github.com/spring-cloud/spring-cloud-contract/commit/de8333deed5f97aca4d8571cbb92941b2e346a45 we do not use the shaded lib, we instead simply exclude the old asm. 

I think we should simply remove the `jsonassert-shade` module as its currently not used and will introduce quite a bit of ripple if used in SCC. 

Thoughts?

Reply:
I think that `jsonassert-shade` makes a lot of sense and I prefer to leave it. If asm doesn't work well for others then we will need to suggest using the shaded module and we will need to rewrite scc a little bit (just the imports).

Reply:
I think it does make a lot of sense if `com.jayway.JsonPath` is not relocated. Because it is exposed by SCC it will probably have a large blast radius for users. When we cross that bridge of incorporating it into SCC we can see what it looks like and we can always adjust it at that point too. 👍

Reply:
Oh @marcingrzejszczak, I stumbled across something yesterday and forgot to share - you may already know about it though. JsonPath has [providers](https://github.com/json-path/JsonPath#jsonprovider-spi) - `json-smart` just happens to be the default. This *could* be another option as well. But anyways - we have beat this one w/ a 🔨  long enough 😸